stable_baselines3/dqn/policies.py

```diff
@@ -0,0 +1,267 @@
+        """
+        Predict the q-values.
+
+        :param obs: (th.Tensor) Observation
+        """
+        features = self.extract_features(obs)
+        return self.q_net(features)
+
+    def _predict(self, observation: th.Tensor, deterministic: bool = False) -> th.Tensor:
+        """
+        Get the action according to the policy for a given observation.
+
+        :param observation: (th.Tensor)
+        :param deterministic: (bool) Whether to use stochastic or deterministic actions
+        :return: (th.Tensor) Taken action according to the policy
+        """
+        # epsilon greedy exploration
+        if not deterministic and np.random.rand() < self.epsilon:
+            action = th.tensor([self.action_space.sample() for _ in range(observation.shape[0])]).reshape(1)
+        else:
+            features = self.extract_features(observation)
+            q_val = self.q_net(features)
+            action = q_val.argmax(dim=1).reshape(-1)
+
+        return action
+
+    def _get_data(self) -> Dict[str, Any]:
+        data = super()._get_data()
+
+        data.update(dict(
+            net_arch=self.net_arch,
+            features_dim=self.features_dim,
+            activation_fn=self.activation_fn,
+            features_extractor=self.features_extractor,
+            epsilon=self.epsilon,
+        ))
+        return data
+
+
+class DQNPolicy(BasePolicy):
+    """
+    Policy class with Q-Value Net and target net for DQN
+
+    :param observation_space: (gym.spaces.Space) Observation space
+    :param action_space: (gym.spaces.Space) Action space
+    :param lr_schedule: (callable) Learning rate schedule (could be constant)
+    :param net_arch: (Optional[List[int]]) The specification of the policy and value networks.
+    :param device: (str or th.device) Device on which the code should run.
+    :param activation_fn: (Type[nn.Module]) Activation function
+    :param epsilon: (float) Exploration fraction (for the epsilon-greedy exploration)
+    :param features_extractor_class: (Type[BaseFeaturesExtractor]) Features extractor to use.
+    :param features_extractor_kwargs: (Optional[Dict[str, Any]]) Keyword arguments
+        to pass to the feature extractor.
+    :param normalize_images: (bool) Whether to normalize images or not,
+         dividing by 255.0 (True by default)
+    :param optimizer_class: (Type[th.optim.Optimizer]) The optimizer to use,
+        ``th.optim.Adam`` by default
+    :param optimizer_kwargs: (Optional[Dict[str, Any]]) Additional keyword arguments,
+        excluding the learning rate, to pass to the optimizer
+    """
+
+    def __init__(self, observation_space: gym.spaces.Space,
+                 action_space: gym.spaces.Space,
+                 lr_schedule: Callable,
+                 net_arch: Optional[List[int]] = None,
+                 device: Union[th.device, str] = 'auto',
+                 activation_fn: Type[nn.Module] = nn.ReLU,
+                 epsilon: float = 0.05,
```

Comment:
I think `epsilon` should be on algorithm level, not in the policy (along with the scheduling stuff). It is not required to use the policy.